### ISSUE TITLE
Anything to do about plt.subplot 

### ISSUE THREAD

**Issue by UboCA** [scverse/scirpy#378]
### Description of the bug

Hi,

I tried to use plt.subplot to merge two `ir.pl.group_abundance` barplot but found them not spaced well.
```
fig, (ax1, ax2) = plt.subplots(1,2, figsize=(4, 4), gridspec_kw={"wspace": 0.6})
ir.pl.group_abundance(adata, groupby='receptor_subtype', target_col='Patients',ax=ax1)
ir.pl.group_abundance(adata, groupby='receptor_subtype', target_col='Tissue',ax=ax2)
```
![image](https://user-images.githubusercontent.com/69028068/212808085-16e6300f-5790-43ea-8b08-62112ee20206.png)

However, while both plotting single object and merging two umap plot, that works.
```
fig, ax1 = plt.subplots(1,1, figsize=(2, 4))
ir.pl.group_abundance(adata, groupby='receptor_subtype', target_col='Patients',ax=ax1)
fig, ax2 = plt.subplots(1,1, figsize=(2, 4))
ir.pl.group_abundance(adata, groupby='receptor_subtype', target_col='Tissue',ax=ax2)
```
![image](https://user-images.githubusercontent.com/69028068/212808213-3bb7540e-c161-4092-85f1-5854a661320f.png)

```
fig, (ax1, ax2) = plt.subplots(1,2, figsize=(8, 3), gridspec_kw={"wspace": 0.6})
sc.pl.umap(adata, color = 'Patients', ax = ax1,show= False)
sc.pl.umap(adata, color = 'Phase', ax= ax2)
```
![image](https://user-images.githubusercontent.com/69028068/212808751-bfc90315-659f-4805-80ee-9a23bbbb2c37.png)

Could you give me any direction about how did that happened?

Best.

### Version information

```txt
scirpy == 0.11.1
scanpy==1.9.1 
anndata==0.8.0 
umap==0.5.3 
numpy==1.21.5 
scipy==1.7.3 
pandas==1.3.5 
scikit-learn==0.24.2 
statsmodels==0.13.2 
python-igraph==0.9.11 
pynndescent==0.5.7
```

**Comment by UboCA** [comment]
I just realized scirpy has used plt.subplot to draw abundance bar plot.
So reusing plt.subplot may not work.

But how could I merge several plot objects from scirpy?

**Comment by UboCA** [comment]
Other observation while I using `ir.pl.group_abundance` is that legend will be resorted by single pair cell number.
```
ir.pl.group_abundance(adata, groupby='chain_pairing', target_col='seurat_clusters', normalize=True, sort 
= 'count')
```
![image](https://user-images.githubusercontent.com/69028068/212812964-47514e5e-a59f-402c-94a1-f3a36c5438a3.png)
You can see seurat_clusters {0...8} are reordered by chain_pairing value.
But what if one want to keep original catgeories? For this reason, to bring an order parametre could make it more friendly.

**Comment by grst** [comment]
Hi @UboCA, 

thanks for reporting this issue. Actually, I'm a bit surprised that this hasn't surfaced earlier. 
With the fix in #379, you should be able to do

```python
adata = ir.datasets.wu2020_3k()
fig, (ax1, ax2) = plt.subplots(1,2, figsize=(12, 4), gridspec_kw={"wspace": 1})
ir.pl.group_abundance(adata, groupby='cluster_orig', target_col='patient',ax=ax1)
ir.pl.group_abundance(adata, groupby='cluster_orig', target_col='source',ax=ax2)
```
![image](https://user-images.githubusercontent.com/7051479/212835320-6794099c-ac43-4eee-b75a-27aa5bc060e4.png)

Could you give it a try? You can install the development version using
```bash
pip install git+https://github.com/scverse/scirpy@fix-subplot
```

---

Regarding your second suggestion, there's a long-outstanding issue for that: https://github.com/scverse/scirpy/issues/186
It should be an easy fix and I'm happy to take a PR. Otherwise I'll look into this once the mega PR #356 is complete.

**Comment by UboCA** [comment]
Got it, many thanks.

Best
